deep_qa/training/text_trainer.py

```diff
@@ -658,12 +659,12 @@ def __get_new_embedding(self, name: str, vocab_name: str='words'):
                 # TimeDistributedEmbedding works with inputs of any shape.
                 embedding_layer = TimeDistributedEmbedding(
                         input_dim=self.data_indexer.get_vocab_size(vocab_name),
-                        output_dim=embedding_params.pop('dimension'),
+                        output_dim=embedding_dim,
                         mask_zero=True,  # this handles padding correctly
                         name=name + '_embedding')
             projection_layer = None
             if embedding_params.pop('project', False):
-                projection_layer = TimeDistributed(Dense(units=embedding_params.pop('dimension'),),
+                projection_layer = TimeDistributed(Dense(units=embedding_dim,),
```

Comment:
I'm confused as to why this was a bug.  Do you really have a case where you project non-pretrained embeddings?  We might actually want to fail here, because you probably don't want to do this.